### PR TITLE
Add Harmonic 🎵

### PR DESCRIPTION
**Pull request by iche033**
Signed-off-by: Ian Chen <ichen@osrfoundation.org>

Add basic information about Gazebo Harmonic. 

I did similar changes as the ones in https://github.com/gazebosim/docs/pull/211 for Garden. The installation instructions were copied from Garden.

Feel free to pick a different emoji if that fits the Harmonic name better :) 

Part of https://github.com/gazebo-tooling/release-tools/issues/856 and https://github.com/gazebo-tooling/release-tools/issues/853

### CODE REVIEW

File: harmonic/install_ubuntu_src.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+```
+QML debugging is enabled. Only use this in a safe environment.
+QML Debugger: Waiting for connection on port 40000...
+```
+
+After that you can just head to
+`QtCreator -> Debug -> Start Debugging -> Attach to QML Port...`
+and enter the QML port
+
+![](images/GzGuiQmlDebugging01.png)
```
Comment (ahcorde) [inline harmonic/install_ubuntu_src.md:246]: where is this image?

File: harmonic/install_ubuntu_src.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+Once you click there, set the port number to 40000 and hit ok
+
+![](images/GzGuiQmlDebugging02.png)
```
Comment (ahcorde) [inline harmonic/install_ubuntu_src.md:250]: where is this image?

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: harmonic/install_ubuntu_src.md (hunk 1 above)
Comment (iche033) [reply harmonic/install_ubuntu_src.md:246]: added images [9a931fd](https://github.com/gazebosim/docs/pull/324/commits/9a931fd599b1f5184266b22a1150090b837eae07)

File: harmonic/install_ubuntu_src.md (hunk 2 above)
Comment (iche033) [reply harmonic/install_ubuntu_src.md:250]: added [9a931fd](https://github.com/gazebosim/docs/pull/324/commits/9a931fd599b1f5184266b22a1150090b837eae07)

File: harmonic/install_osx_src.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+their repositories:
+
+```bash
+vcs import < collection-harmonic.yaml
+```
+
+The src subdirectory should contain all the sources ready to be built.
+
+
+## Install dependencies
```
Comment (ahcorde) [inline harmonic/install_osx_src.md:71]:
```suggestion
The src subdirectory should contain all the sources ready to be built.

## Install dependencies
```

File: harmonic/install_osx_src.md [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+```
+
+General dependencies:
+
+```bash
+brew install assimp boost bullet cmake cppzmq dartsim@6.10.0 doxygen eigen fcl ffmpeg flann freeimage freetype gflags google-benchmark gts ipopt irrlicht jsoncpp libccd libyaml libzzip libzip nlopt ode open-scene-graph ossp-uuid ogre1.9 ogre2.2 pkg-config protobuf qt qwt rapidjson ruby tbb tinyxml tinyxml2 urdfdom zeromq
+```
+
+`dartsim@6.10.0` and `qt5` are not sym-linked. To use those dependencies when building
+`gazebo-physics2` and `gazebo-gui3`, run the following after installation to add them to `/use/local`:
```
Comment (ahcorde) [inline harmonic/install_osx_src.md:95]: remove this?

File: harmonic/install_ubuntu_src.md [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+
+To build a single package:
+
+```bash
+colcon build --packages-select PACKAGE_NAME
+```
+
+Visit [colcon documentation](https://colcon.readthedocs.io/en/released/#) to view more `colcon` build and test options.
+
+If there are no errors, all the binaries should be ready to use. You can check the [Troubleshooting] section for errors.
```
Comment (ahcorde) [inline harmonic/install_ubuntu_src.md:175]: link to Troubleshooting section
```suggestion
If there are no errors, all the binaries should be ready to use. You can check the [Troubleshooting](/docs/Harmonic/troubleshooting#ubuntu) section for errors.
```

File: harmonic/install_windows_src.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+```bash
+colcon graph
+```
+
+`colcon graph` should list the Gazebo libraries with an
+[interdependency diagram](https://colcon.readthedocs.io/en/released/reference/verb/graph.html#example-output).
+If that is the case, then you are ready to build the whole set of libraries:
+
+```bash
+colcon build --cmake-args -DBUILD_TESTING=OFF --merge-install --packages-up-to gz-sim7
```
Comment (ahcorde) [inline harmonic/install_windows_src.md:90]: According with the [collection-garden.yaml](https://raw.githubusercontent.com/ignition-tooling/gazebodistro/master/collection-harmonic.yaml) file Gazebo version is the one in `main` who corresponds with `gz-sim8`

File: harmonic/troubleshooting.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+I cannot find any available 'gz' command:
+	* Did you install any gazebo library?
+	* Did you set the GZ_CONFIG_PATH environment variable?
+	    E.g.: export GZ_CONFIG_PATH=$HOME/local/share/gazebo
+```
+
+You should set up the environment variable `GZ_CONFIG_PATH=/usr/local/share/gazebo/`
+
+
+## macOS
```
Comment (ahcorde) [inline harmonic/troubleshooting.md:16]:
```suggestion
You should set up the environment variable `GZ_CONFIG_PATH=/usr/local/share/gazebo/`

## macOS
```

File: harmonic/troubleshooting.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+
+# Source ~/.bashrc in terminal
+. ~/.bashrc
+```
+
+### No rule to make target `'/usr/lib/libm.dylib', needed by 'lib/libgz-physics6-dartsim-plugin.6.1.0.dylib'. Stop.`
+Try to run `brew outdated` followed by a `brew upgrade` may fix some of it.
+
+
+## Ubuntu
```
Comment (ahcorde) [inline harmonic/troubleshooting.md:73]:
```suggestion
### No rule to make target `'/usr/lib/libm.dylib', needed by 'lib/libgz-physics6-dartsim-plugin.6.1.0.dylib'. Stop.`
Try to run `brew outdated` followed by a `brew upgrade` may fix some of it.

## Ubuntu
```

File: harmonic/troubleshooting.md [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+drivers.
+
+The Ogre 2 debs from the osrfoundation repository are built from a fork of
+Ogre's `v2-1` branch with changes needed for deb packaging and allowing it to
+be co-installable with Ogre 1.x. The code can be found here:
+
+https://github.com/gazebo-forks/ogre-2.1-release
+
+
+## Windows
```
Comment (ahcorde) [inline harmonic/troubleshooting.md:152]:
```suggestion
https://github.com/gazebo-forks/ogre-2.1-release

## Windows
```

**Review by ahcorde — DISMISSED** [review]

File: harmonic/install_osx_src.md (hunk 3 above)
Comment (iche033) [reply harmonic/install_osx_src.md]: [5f8e3e2](https://github.com/gazebosim/docs/pull/324/commits/5f8e3e20eb115262e864f8d151a85f3cf7852096)

File: harmonic/install_osx_src.md (hunk 4 above)
Comment (iche033) [reply harmonic/install_osx_src.md]: actually I had to do this on my mac. I expanded the instructions to include paths for arm-based Apple silicon processors. [5f8e3e2](https://github.com/gazebosim/docs/pull/324/commits/5f8e3e20eb115262e864f8d151a85f3cf7852096)

File: harmonic/install_ubuntu_src.md (hunk 5 above)
Comment (iche033) [reply harmonic/install_ubuntu_src.md]: [5f8e3e2](https://github.com/gazebosim/docs/pull/324/commits/5f8e3e20eb115262e864f8d151a85f3cf7852096)

File: harmonic/troubleshooting.md (hunk 8 above)
Comment (iche033) [reply harmonic/troubleshooting.md]: [5f8e3e2](https://github.com/gazebosim/docs/pull/324/commits/5f8e3e20eb115262e864f8d151a85f3cf7852096)

File: harmonic/troubleshooting.md (hunk 9 above)
Comment (iche033) [reply harmonic/troubleshooting.md]: [5f8e3e2](https://github.com/gazebosim/docs/pull/324/commits/5f8e3e20eb115262e864f8d151a85f3cf7852096). Also fixed ogre version, 2.1 -> 2.3

File: harmonic/install_windows_src.md (hunk 6 above)
Comment (iche033) [reply harmonic/install_windows_src.md]: [5f8e3e2](https://github.com/gazebosim/docs/pull/324/commits/5f8e3e20eb115262e864f8d151a85f3cf7852096)

File: harmonic/troubleshooting.md [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+	* Did you install any Gazebo library?
+	* Did you set the GZ_CONFIG_PATH environment variable?
+	    E.g.: export GZ_CONFIG_PATH=$HOME/local/share/gz
+```
+
+You should set up the environment variable:
+
+```
+# replace <path_to_install_dir> to your Gazebo installation directory
+GZ_CONFIG_PATH=<path_to_install_dir>/share/gz/
```
Comment (azeey) [inline harmonic/troubleshooting.md:17]: The garden version has `GZ_CONFIG_PATH=/usr/local/share/gazebo/`. Should that be updated as well?

File: harmonic/troubleshooting.md [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+
+If that loads, you can continue to use Gazebo Sim with Ogre 1, just use the
+`--render-engine ogre` option.
+
+To enable Ogre 2 support, you'll need to update your computer's OpenGL version.
+As suggested on the Ogre logs, this may require updating your graphics card
+drivers.
+
+The Ogre 2 debs from the osrfoundation repository are built from a fork of
+Ogre's `v2-3` branch with changes needed for deb packaging and allowing it to
```
Comment (azeey) [inline harmonic/troubleshooting.md:149]: The garden version has `v2-1`. I think we need to update that as well.

File: harmonic/troubleshooting.md (hunk 10 above)
Comment (iche033) [reply harmonic/troubleshooting.md:17]: yep updated and ported more fixes to Garden in [937916a](https://github.com/gazebosim/docs/pull/324/commits/937916a493ee0ed5f721e5ab648f62c5d4717a3d)

File: harmonic/troubleshooting.md (hunk 11 above)
Comment (iche033) [reply harmonic/troubleshooting.md:149]: done. [937916a](https://github.com/gazebosim/docs/pull/324/commits/937916a493ee0ed5f721e5ab648f62c5d4717a3d)

**Pull request merged**